test/shell-pipeline-tests/common.sh

```diff
@@ -78,7 +78,7 @@ function createImageRegistrySecret() {
 function usePRBranchInPipelineRunIfPRCheck() {
   local -r PIPELINE_RUN_FILE_PATH=$1
 
-  if [ -n $PULL_NUMBER ]; then
+  if [ -n "$PULL_NUMBER" ]; then
```

Comment:
One thing to consider is to have `shellcheck` for these shell scripts. Here's what it would have printed for this specific line:

```
In test/shell-pipeline-tests/common.sh line 81:
  if [ -n $PULL_NUMBER ]; then
          ^----------^ SC2070 (error): -n doesn't work with unquoted arguments. Quote or use [[ ]].
          ^----------^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
  if [ -n "$PULL_NUMBER" ]; then
```